ting_subsearch_secondary.module

```diff
@@ -176,7 +192,7 @@ function ting_subsearch_secondary_execute_secondary_search($keys, $results) {
   if ($secondary_num_objects > $secondary_result_min && $secondary_num_objects < $secondary_max) {
     $ratio = $secondary_num_objects / $num_total_objects;
 
-    if ($ratio > variable_get('ting_subsearch_secondary_ps_factor', 2)) {
+    if ($ratio > ting_subsearch_common_tofloat(variable_get('ting_subsearch_secondary_ps_factor', 2))) {
```

Comment:
All of these `ting_subsearch_translate_ps_factor` seems like a hacky way of manipulating data in background.
I assume these variable settings come from some sort of admin, so I would suggest to simply create a validation mechanism for floating point values.
And a hook_update to transform existing ones to proper format.

Reply:
Added `hook_update` and custom form element validation in `ting_subsearch_common` to ensure that the correct float values separator is used.